models/kendra.go

```diff
@@ -0,0 +1,65 @@
+package models
+
+import (
+	"context"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/service/kendra"
+)
+
+var opts = kendra.Options{
+	Credentials: prov,
+	Region:      region,
+}
+
+var client = kendra.New(opts)
+
+type KendraResult struct {
+	ImgPath string
+	Title   string
+	Link    string
+	Excerpt string
+}
+
+type KendraResults struct {
+	QueryOutput kendra.QueryOutput
+	Results     []KendraResult
+	Query       string
+	Count       int
+}
+
+func queryOutputToResults(out kendra.QueryOutput) KendraResults {
+	results := KendraResults{
+		Results: []KendraResult{},
+	}
+
+	for _, item := range out.ResultItems {
+		res := KendraResult{
+			ImgPath: "images/placeholder.jpg",
+			Title:   *item.DocumentTitle.Text,
+			Excerpt: *item.DocumentExcerpt.Text,
+			Link:    *item.DocumentURI,
+		}
+		results.Results = append(results.Results, res)
+		results.Count = int(*out.TotalNumberOfResults)
+	}
+
+	return results
+}
+
+func MakeQuery(query string) KendraResults {
+	kendra_query := kendra.QueryInput{
+		IndexId:   &index_id,
+		QueryText: &query,
+	}
+	out, err := client.Query(context.TODO(), &kendra_query)
+
+	// WARN: this needs to be fixed to a proper error
```

Comment:
Ok you can make this into a TODO and i'll give someone a task to work on this

Reply:
the plugin i use for Neovim has the warnings be orange and i like it more than the todo color 🙃. Ill change it.